app/lib/actions/katello/repository/destroy.rb

```diff
@@ -16,6 +16,7 @@ def plan(repository, options = {})
           plan_action(Pulp::Repository::Destroy, pulp_id: repository.pulp_id)
           plan_action(Product::ContentDestroy, repository) unless skip_environment_update
           plan_action(ElasticSearch::Repository::Destroy, pulp_id: repository.pulp_id)
+          plan_action(Katello::Repository::DestroyMedia, repository)
```

Comment:
nah, as long as its in the plan phase its fine, if it errors there will be a full AR rollback